esp/public/media/theme_editor/theme_editor.js

```diff
@@ -54,40 +54,47 @@ $j(document).ready(function(){
     showColor();
     showBasePalette();
     showCustomPalette();
-    $j('#addToPalette').click(function(){
-        var newColor = $j('<input>');
-        newColor.addClass('palette');
-        newColor.attr({
-            name: 'palette',
-            type: 'text',
-            value: 'black'
-        });
-        $j('#palette_custom_div').append(newColor);
-        $j('#palette_custom_div').append(" ");
-        showCustomPalette();
-    });
-
+    
     $j('[rel=popover]').each(function(){
         $j(this).popover({placement:'left', animation:false});
     });
     
-    //  Show optional variable selector if there are optional variables available
     $j('div.opt_var_div:has(select.select_opt_var:has(option))').removeClass('hidden');
     
-    //  Allow form elements to be created for optional variables
     $j('button.add_opt_var_button').click(function (event) {
         event.preventDefault();
         
-        //  Determine which optional variable we are trying to add.
+        // Determine which optional variable we are trying to add.
         var button_id = event.target.id;
         var button_id_prefix = 'add_opt_var_';
         var select_id = 'new_opt_var_' + button_id.substr(button_id_prefix.length);
         var select_val = $j('#' + select_id).val();
+        
+        // Create a new color input field
+        var colorInput = $j('<input>')
+            .attr({
+                id: 'id_' + select_val,
+                class: 'color',
+                type: 'text',
+                value: '#000000',
+                name: select_val,
+                style: 'display: none;'
+            });
 
-        //  Insert a new color selector for the desired variable
-        $j(event.target).parent().parent().parent().append($j('<div class="control-group">')
+        // Create the "Reset Color" button
+        var resetButton = $j('<button class="reset-color">Reset Color</button>').click(function() {
+            colorInput.spectrum("set", "#000000"); // Reset color to default
+        });
```

Comment:
Adding this button is great, but I think we really want a way to remove the optional variables, not to just reset them to black.